tools/sloop.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * Library of functions pertaining to Sloop shadows. Slooping is a way
+ * to simulate the built-in snooping functionality of EotL.
+ *
+ * @author gon
+ */
+
+#define SLOOPEE_SHAD "/usr/gon/tools/sloopee_shad"
+#define SLOOPER_SHAD "/usr/gon/tools/slooper_shad"
+
+// Function Declarations
+protected void add_sloop_shadow( object ob, string str );
+protected void dest_sloop_shadows( object ob, string str );
+protected object * query_sloop_shadows( object ob, string str );
+void add_slooper_shadow( object ob );
+void add_sloopee_shadow( object ob );
+void dest_slooper_shadow( object ob );
+void dest_sloopee_shadow( object ob );
+object find_slooper_shadow( object ob );
+object find_sloopee_shadow( object ob );
+
+/**
+ * Function filters all shdow objects and returns a filtered array of shadows
+ * matching either SLOOPEE_SHAD or SLOOPER_SHAD
+ * @param  object ob   Object passed to check for shadow
+ * @param  string str    string to match (Either SLOOPER_SHAD or SLOOPEE_SHAD)
+ * @return   array of matching shadow(s)
+ */
+protected object * query_sloop_shadows( object ob, string str ) {
+    object * shadow_ob = shadow_list( ob );
+    return filter( shadow_ob, (: return ( program_name( $1 ) == $2 ); :), str );
```

Comment:
The one question I have, is whether or not this should apply at all times or simply when we bump into the 80 character limit?